modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/result/AbstractReturnDelegate.java

```diff
@@ -293,13 +304,36 @@ protected MediaType resolveMediaType(Object value, MediaType contextMimeType, Ch
     return contextMimeType.withCharset(contextEncoding);
   }
 
+  private Runnable getIncrementActiveComponent(ExecutionContextAdapter executionContext) {
+    Optional<ConfigurationInstance> config = executionContext.getConfiguration();
+    if (config.isPresent()) {
+      return () -> tryToMutateConfigurationStats(config.get(), (MutableConfigurationStats::addActiveComponent));
+    }
+    return null;
+  }
+
+  private Runnable getDecrementActiveComponent(ExecutionContextAdapter executionContext) {
+    Optional<ConfigurationInstance> config = executionContext.getConfiguration();
+    if (config.isPresent()) {
+      return () -> tryToMutateConfigurationStats(config.get(), (MutableConfigurationStats::discountActiveComponent));
+    }
+    return null;
+  }
+
   protected class ConnectedInputStreamWrapper extends ProxyInputStream {
 
     private final ConnectionHandler<?> connectionHandler;
+    private final Runnable onClose;
+    private AtomicBoolean alreadyClosed = new AtomicBoolean(false);
 
-    private ConnectedInputStreamWrapper(InputStream delegate, ConnectionHandler<?> connectionHandler) {
+    private ConnectedInputStreamWrapper(InputStream delegate, ConnectionHandler<?> connectionHandler, Runnable onCreate,
+                                        Runnable onClose) {
       super(delegate);
       this.connectionHandler = connectionHandler;
+      this.onClose = onClose;
+      if (onCreate != null) {
```

Comment:
this doesn't make sense. If you want this task executed when the instance is created, then call it after doing the `new`... This should not be the constructor's responsibility